File: servidor/modelo/RegistroIA.cpp
#include "RegistroIA.h"

using namespace std;

RegistroIA::RegistroIA(int idEsp32, const string& tipo, double confianza) {
    this->idEsp32 = idEsp32;
    this->tipo = tipo;
    this->confianza = confianza;
}

int RegistroIA::getIdEsp32()     { return idEsp32; }
string RegistroIA::getTipo(){ return tipo; }
double RegistroIA::getConfianza(){ return confianza; }

File: flutter_app/lib/modelo/RegistroIA.dart
/// Detección de la IA, tal como la envía el servidor:
/// "id_camara;fecha_hora;tipo;confianza".
class RegistroIA {
  final String idCamara;
  final String fecha;
  final String tipo;
  final String confianza;

  RegistroIA(this.idCamara, this.fecha, this.tipo, this.confianza);

  factory RegistroIA.desdeLinea(String linea) {
    final p = linea.split(';');
    return RegistroIA(p[0], p[1], p[2], p[3]);
  }
}
